src/all/namicomi/src/eu/kanade/tachiyomi/extension/all/namicomi/NamiComi.kt

```diff
@@ -0,0 +1,381 @@
+package eu.kanade.tachiyomi.extension.all.namicomi
+
+import android.app.Application
+import android.content.SharedPreferences
+import androidx.preference.ListPreference
+import androidx.preference.PreferenceScreen
+import androidx.preference.SwitchPreferenceCompat
+import eu.kanade.tachiyomi.extension.all.namicomi.dto.ChapterListDto
+import eu.kanade.tachiyomi.extension.all.namicomi.dto.CoverArtDto
+import eu.kanade.tachiyomi.extension.all.namicomi.dto.EntityAccessMapDto
+import eu.kanade.tachiyomi.extension.all.namicomi.dto.EntityAccessRequestDto
+import eu.kanade.tachiyomi.extension.all.namicomi.dto.EntityAccessRequestItemDto
+import eu.kanade.tachiyomi.extension.all.namicomi.dto.MangaDto
+import eu.kanade.tachiyomi.extension.all.namicomi.dto.MangaListDto
+import eu.kanade.tachiyomi.extension.all.namicomi.dto.PageListDto
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import okhttp3.CacheControl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import okio.IOException
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+
+abstract class NamiComi(final override val lang: String, private val extLang: String = lang) :
+    ConfigurableSource, HttpSource() {
+
+    override val name = "NamiComi"
+    override val baseUrl = NamiComiConstants.webUrl
+    override val supportsLatest = true
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    private val helper = NamiComiHelper(lang)
+
+    final override fun headersBuilder() = super.headersBuilder().apply {
+        set("Referer", "$baseUrl/")
+        set("Origin", baseUrl)
+    }
+
+    override val client = network.client.newBuilder()
+        .rateLimit(3)
+        .addNetworkInterceptor { chain ->
+            val response = chain.proceed(chain.request())
+
+            if (response.code == 402) {
+                throw IOException(helper.intl["error_payment_required"])
+            }
```

Comment:
```suggestion
            if (response.code == 402) {
                response.close()
                throw IOException(helper.intl["error_payment_required"])
            }
```
need to close the response as well, I forgot about it before